### ISSUE TITLE
False positive on unused import in type annotation

### ISSUE THREAD

**Issue by charliermarsh** [charliermarsh/ruff#376]
```py
import sys

import numpy as np

if sys.version_info >= (3, 10):
    from typing import TypeAlias
else:
    from typing_extensions import TypeAlias


CustomInt: TypeAlias = "np.int8 | np.int16"
```

**Comment by charliermarsh** [comment]
\cc @ghuls

**Comment by ghuls** [comment]
When there is a `from __future__ import annotations` line before, it doesn't work yet.

```python
from __future__ import annotations

import sys

import numpy as np

if sys.version_info >= (3, 10):
    from typing import TypeAlias
else:
    from typing_extensions import TypeAlias


CustomInt: TypeAlias = "np.int8 | np.int16"
```

**Comment by charliermarsh** [comment]
Good call. Will fix that too.

**Comment by charliermarsh** [comment]
Should be resolved by #378.

**Comment by ghuls** [comment]
Is fixed indeed! Thanks.
